### PR TITLE
xtensa: polish doxygen and add to missing doc

### PR DESCRIPTION
**Pull request by dcpleung**
(See individual commits for details.)

### CODE REVIEW

**Review by nashif — DISMISSED** [review]

**Review by kartben — COMMENTED** [review]
See some first feedback. I need to have a closer look at the "doc: force include interface header files first" commit as it feels weird that we would need this stuff (but it looks like you've spent some time investigating, so I certainly don't want to imply that this might not be the solution).

File: include/zephyr/arch/xtensa/gdbstub.h [hunk 1]
```diff
@@ -35,32 +35,35 @@
...
 	 */
 	int8_t		stack_offset;
 
-	/* Sequence number */
+	/** Sequence number */
 	uint8_t		seqno;
 
-	/* Set 1 to if register should not be written
+	/**
+	 * Set 1 to if register should not be written
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply include/zephyr/arch/xtensa/gdbstub.h]: Fixed.

File: include/zephyr/arch/xtensa/gdbstub.h [hunk 2]
```diff
@@ -17,8 +17,8 @@
 #define XTREG_GRP_SPECIAL		0x0200
 #define XTREG_GRP_USER			0x0300
 
-/*
- * Register description fot GDB stub.
+/**
+ * @brief Register description fot GDB stub.
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply include/zephyr/arch/xtensa/gdbstub.h]: Fixed.

File: include/zephyr/sys/atomic.h [hunk 3]
```diff
@@ -245,6 +245,255 @@ static inline void atomic_set_bit_to(atomic_t *target, int bit, bool val)
...
+/**
+ * @brief Atomic compare-and-set.
+ *
+ * This routine performs an atomic compare-and-set on @a target. If the current
+ * value of @a target equals @a old_value, @a target is set to @a new_value.
+ * If the current value of @a target does not equal @a old_value, @a target
+ * is left unchanged.
+ *
+ * @note As for all atomic APIs, includes a
+ * full/sequentially-consistent memory barrier (where applicable).
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply include/zephyr/sys/atomic.h]: Thanks. Replaced them with the alias.

**Comment by dcpleung** [comment]
> See some first feedback. I need to have a closer look at the "doc: force include interface header files first" commit as it feels weird that we would need this stuff (but it looks like you've spent some time investigating, so I certainly don't want to imply that this might not be the solution).

AFAICT, if you give it a directory, doxygen parses files in alphabetically order. This means that whichever file has the first definition of a symbol is where the link points to. For example, the arch interfaces in `sys/arch_interface.h` are implemented in various architectures under `arch/*`. So the link points to the ones in `arch/*` instead of those under `sys/arch_interface.h`. So we need to force doxygen to parse the files in certain order to force it to link correctly.

**Review by nashif — DISMISSED** [review]

**Comment by nashif** [comment]
@dcpleung  please rebase.

**Comment by dcpleung** [comment]
> @dcpleung please rebase.

Done.

**Review by nashif — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**